x-pack/plugins/maps/public/connected_components/layer_panel/layer_settings/layer_settings.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ChangeEvent, Fragment } from 'react';
+import {
+  EuiTitle,
+  EuiPanel,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { MAX_ZOOM } from '../../../../common/constants';
+import { AlphaSlider } from '../../../components/alpha_slider';
+import { ValidatedDualRange } from '../../../../../../../src/plugins/kibana_react/public';
+import { ILayer } from '../../../classes/layers/layer';
+
+interface Props {
+  layer: ILayer | null;
+  updateLabel: (layerId: string, label: string) => void;
+  updateMinZoom: (layerId: string, minZoom: number) => void;
+  updateMaxZoom: (layerId: string, maxZoom: number) => void;
+  updateAlpha: (layerId: string, alpha: number) => void;
+  updateLabelsOnTop: (layerId: string, labelsOnTop: boolean) => void;
+}
+
+export function LayerSettings(props: Props) {
+  if (!props.layer) {
```

Comment:
wouldn't this make the `!.` syntax unnecessary?

Reply:
No, since they are in different scope because they are in functions. I have updated the component to make the signagure for layer to be `ILayer` instead of `ILayer | null` to avoid all the fuse.